rumqttd/src/server/tls.rs

```diff
@@ -200,39 +200,42 @@ impl TLSAcceptor {
             // Get certificates
             let cert_file = File::open(cert_path);
             let cert_file = cert_file.map_err(|_| Error::ServerCertNotFound(cert_path.clone()))?;
-            let certs = rustls_pemfile::certs(&mut BufReader::new(cert_file));
-            let certs = certs.map_err(|_| Error::InvalidServerCert(cert_path.to_string()))?;
-            let certs = certs
-                .iter()
-                .map(|cert| Certificate(cert.to_owned()))
-                .collect();
+            let certs = rustls_pemfile::certs(&mut BufReader::new(cert_file))
+                .collect::<Result<Vec<_>, _>>()
+                .map_err(|_| Error::InvalidServerCert(cert_path.to_string()))?;
 
             // Get private key
             let key = first_private_key_in_pemfile(key_path)?;
 
             (certs, key)
         };
 
-        let builder = ServerConfig::builder().with_safe_defaults();
+        let builder = ServerConfig::builder();
 
         // client authentication with a CA. CA isn't required otherwise
         #[cfg(feature = "verify-client-cert")]
         let builder = {
             let ca_file = File::open(ca_path);
             let ca_file = ca_file.map_err(|_| Error::CaFileNotFound(ca_path.clone()))?;
             let ca_file = &mut BufReader::new(ca_file);
-            let ca_certs = rustls_pemfile::certs(ca_file)?;
+            let ca_certs = rustls_pemfile::certs(ca_file).collect::<Result<Vec<_>, _>>()?;
             let ca_cert = ca_certs
-                .first()
-                .map(|c| Certificate(c.to_owned()))
+                .into_iter()
+                .next()
```

Comment:
if we only need the _first_ one, shall we really check if all other certs are valid are not? ( this is what `collect` does right? )

Reply:
That's what the existing code is doing, but okay.

Reply:
> That's what the existing code is doing, but okay.
>

right, but I just wanted to know from users perspective, what would one expect it to work like, will only one ( in this care, the first one ) being correct and other can be whatever should work, or whole file must be correct for it to work?

that is why I asked it :sweat_smile: wdyt?

Reply:
also, as per this [comment](https://github.com/bytebeamio/rumqtt/pull/795#discussion_r1489471752), would it be better to club this change with that follow-up PR?

Reply:
ah okie, then current state is fine.